source/interface/directoryContents.js

```diff
@@ -4,14 +4,12 @@ const { merge } = require("../merge.js");
 const { handleResponseCode, processResponsePayload } = require("../response.js");
 const { normaliseHREF, normalisePath } = require("../url.js");
 const { getSingleValue, getValueForKey, parseXML, propsToStat } = require("./dav.js");
-const { encodePath, prepareRequestOptions, request } = require("../request.js");
+const { appendSlash, encodePath, prepareRequestOptions, request } = require("../request.js");
 
-function getDirectoryContents(remotePathRaw, options) {
-    // Strip the ending slash
-    const remotePath = remotePathRaw.replace(/\/$/, "");
+function getDirectoryContents(remotePath, options) {
     // Join the URL and path for the request
     const requestOptions = {
-        url: joinURL(options.remoteURL, encodePath(remotePath)),
+        url: appendSlash(joinURL(options.remoteURL, encodePath(appendSlash(remotePath)))),
```

Comment:
This could actually be `joinURL(options.remoteURL, encodePath(remotePath), "/")`

Reply:
That was my initial approach, but I thought you asked for a helper function perhaps for clarity. (Sorry for the dup function btw)